src/apps/parseBlocks/parser/model.ts

```diff
@@ -68,8 +68,10 @@ export function optional<T, TContext extends Context>(
 }
 
 // look for 0 or more of something, until we can't parse any more. note that this function never fails, it will instead succeed with an empty array.
+// function will fail when param mayFail is set to true
 export function many<T, TContext extends Context>(
-  parser: Parser<T, TContext>
+  parser: Parser<T, TContext>,
+  failWhenNoResult?: boolean
```

Comment:
I think it would be more clear to create a separate function `oneOrMore`. It can be hard to tell on the calling side what does `many(parser, true)` mean.